readthedocs/api/v2/serializers.py

```diff
@@ -127,11 +129,33 @@ class VersionAdminSerializer(VersionSerializer):
 class BuildCommandSerializer(serializers.ModelSerializer):
 
     run_time = serializers.ReadOnlyField()
+    command = serializers.SerializerMethodField()
 
     class Meta:
         model = BuildCommandResult
         exclude = []
 
+    def get_command(self, obj):
+        # HACK: remove unreadable paths from the command outputs when returning it from the API.
+        # We could make this change at build level, but we want to avoid undoable issues from now
+        # and hack a small solution to fix the immediate problem.
+        #
+        # This converts:
+        #   $ /usr/src/app/checkouts/readthedocs.org/user_builds/<container_hash>/<project_slug>/envs/<version_slug>/bin/python
+        #   $ /home/docs/checkouts/readthedocs.org/user_builds/<project_slug>/envs/<version_slug>/bin/python
+        # into
+        #   $ python
+        project_slug = obj.build.version.project.slug
+        version_slug = obj.build.version.slug
+
+        docroot = settings.DOCROOT
+        # Remove Docker hash from DOCROOT when running it locally
+        if settings.RTD_DOCKER_COMPOSE:
+            docroot = re.sub('/[0-9a-z]+/?$', '', settings.DOCROOT, count=1)
+
+        command = re.sub(f'{docroot}/[0-9a-z]+/{project_slug}/envs/{version_slug}(/bin/)?', '', obj.command, count=1)
```

Comment:
```suggestion
        command = re.sub(f'{docroot}/{project_slug}/envs/{version_slug}(/bin/)?', '', obj.command, count=1)
```

Reply:
This is definitely nicer and more explicit.

Reply:
I'm not sure I understand the change. `[0-9a-z]+/` was removed. But if this regex worked _before_ isn't it broken now?

Reply:
Oh, I thought we were changing `[0-9a-z]+` to `project_slug`, but I see we just removed it. Not 100% sure.

Reply:
I added this suggestion to not forget to review this on Monday, basically.

I added `[0-9a-z]+` to remove the "container hash" we append on the docroot when working locally to avoid collision between builders when using more than 1 builder. However, if you take a look at line 154 and 156 both include that part of the regex when it should only be on line 154 (only when working locally)

Reply:
OK, my assumption when I saw this code again was wrong.

Now, I remember why I've done this:

- the docroot _contains_ the containers' hash and we have to remove it because it is different every time spin up Docker again. So, it can't be part of the regex
- in the regex _we need_ that part `[0-9a-z]` because it will match the paths that are already in the commands

So, we need both and the code was correct as it was.